lib/firmata.js

```diff
@@ -541,13 +540,7 @@ var Board = function(port, options, callback) {
   }.bind(this));
 
   this.transport.on("data", function(data) {
-    var byt, cmd;
-
-    if (!this.versionReceived && data[0] !== REPORT_VERSION) {
```

Comment:
When I was tinkering yesterday I was thinking about this as well, I'm glad you had time to address it directly